.github/workflows/staging-builds.yml

```diff
@@ -1,17 +1,22 @@
-name: Build transformer release image
+name: Build transformer image for staging env
+
 on:
-  push:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
     branches:
-      - 'rudder-saas'
+      - main
```

Comment:
This trigger is incorrect. We need to update staging environment only when pull requests are raised from the release candidate branches.

Reply:
added conditional check on to trigger only when branch name starts with release or hotfix